seahorn/CMakeLists.txt

```diff
@@ -300,7 +301,7 @@ endfunction()
 set(MAX_LINKED_LIST_ITEM_ALLOCATION_SIZE 5)
 set(MAX_BUFFER_SIZE 10)
 set(MAX_STRING_LEN 16)
-set(MAX_INITIAL_SIZE 10)
+set(MAX_INITIAL_SIZE 4096)
```

Comment:
This is the bound size used for the array list. Here is what we used for the bounded array list:
https://github.com/seahorn/verify-c-common/blob/0401c5d0026a8f191ea9b78f9b3bba73bbbc1b0f/seahorn/lib/array_list_helper.c#L29
I assumed we used 4k as the maximum memory allocation. Do we have to explicitly make this assumption for seahorn?

Reply:
We did not have such a restriction before. We do not want to have it now.

It might be necessary for KLEE, which is ok, but it should be ignored for SeaHorn and properly documented.

Reply:
You can always use `SIZE_MAX` for it for seahorn

Reply:
Could I do the following:
1. Still use 4k as the maximum memory allocation for KLEE.
2. Will delete assumption on this file for SeaHorn: https://github.com/seahorn/verify-c-common/blob/0401c5d0026a8f191ea9b78f9b3bba73bbbc1b0f/seahorn/lib/array_list_helper.c#L29
3. Maybe 2. is not a better choice. May I change it as`assume(list->current_size <= SIZE_MAX);` for SeaHorn?

Reply:
the assumption that you are suggesting will be compiled to nothing. So it will be bad to have in SeaHorn. Better remove it. 

You can do whatever you need for KLEE, just document this and make sure not to mix KLEE specifics with SeaHorn specifics.

Reply:
Got it! Thanks for the tips.